awx/main/dispatch/worker/callback.py

```diff
@@ -26,7 +30,37 @@ class CallbackBrokerWorker(BaseWorker):
 
     MAX_RETRIES = 2
 
+    def __init__(self):
+        self.buff = {}
+        self.last_write = tz_now()
+
+    def read(self, queue):
+        try:
+            return queue.get(block=True, timeout=.25)
+        except QueueEmpty:
+            return 'FLUSH'
+
+    def flush(self):
+        should_flush = (
+            any([len(events) == 2000 for events in self.buff.values()]) or
+            (tz_now() - self.last_write).seconds > .1
+        )
+        if should_flush:
+            now = tz_now()
+            for cls, events in self.buff.items():
+                for e in events:
+                    e.created = e.modified = now
+                cls.objects.bulk_create(events)
+                for e in events:
+                    post_save.send(e.__class__, instance=e, created=True)
```

Comment:
I would find it more clear if we could just have this defined as our own method (and call that method), without going through the Django signals connection. We should never rely on Django firing the job event related signals - thus we shouldn't connect them in the first place.

Reply:
Addressed.